diff -urpN qgit/src/common.h qgit.new/src/common.h
--- qgit/src/common.h	2009-03-12 18:14:31.000000000 +0100
+++ qgit.new/src/common.h	2009-05-15 23:14:10.000000000 +0200
@@ -29,13 +29,13 @@ inline const QString& _valueOf(const QSt
 inline const QString  _valueOf(size_t x) { return QString::number((uint)x); }
 
 // some debug macros
-#define dbg(x)    qDebug(#x " is <%s>", _valueOf(x).toLatin1().constData())
-#define dbs(x)    qDebug(_valueOf(x).toLatin1().constData())
-#define dbp(s, x) qDebug(QString(s).arg(_valueOf(x)).toLatin1().constData())
-#define db1       qDebug("Mark Nr. 1")
-#define db2       qDebug("Mark Nr. 2")
-#define db3       qDebug("Mark Nr. 3")
-#define db4       qDebug("Mark Nr. 4")
+#define dbg(x)    qDebug("%s", #x " is <%s>", _valueOf(x).toLatin1().constData())
+#define dbs(x)    qDebug("%s", _valueOf(x).toLatin1().constData())
+#define dbp(s, x) qDebug("%s", QString(s).arg(_valueOf(x)).toLatin1().constData())
+#define db1       qDebug("%s", "Mark Nr. 1")
+#define db2       qDebug("%s", "Mark Nr. 2")
+#define db3       qDebug("%s", "Mark Nr. 3")
+#define db4       qDebug("%s", "Mark Nr. 4")
 #define dbStart   dbs("Starting timer..."); QTime _t; _t.start()
 #define dbRestart dbp("Elapsed time is %1 ms", _t.restart())
 
diff -urpN qgit/src/exceptionmanager.cpp qgit.new/src/exceptionmanager.cpp
--- qgit/src/exceptionmanager.cpp	2007-12-21 20:23:07.000000000 +0100
+++ qgit.new/src/exceptionmanager.cpp	2009-05-15 23:14:56.000000000 +0200
@@ -35,7 +35,7 @@ bool ExceptionManager::isMatch(int value
 	if (match) {
 		QString info("Caught exception \'" + descriptions[excp] +
 		             "\' while in " + context);
-		qDebug(info.toLatin1().constData());
+		qDebug("%s", info.toLatin1().constData());
 	}
 	return match;
 }
